### PR TITLE
Fr/openapi spec fix

### PR DESCRIPTION
**Pull request by furqanrydhan**
## Changes

- Fix OpenAPI errors

## How this PR will be tested

- [ ] test contract read and write functions to ensure they work, confirm types are correct and no use cases stop working

## Output

---

## PR-Codex overview
This PR focuses on updating TypeScript type imports, enhancing schemas using `@sinclair/typebox`, and adding configuration for `biome`. It also modifies response structures in various routes and adjusts the `.gitignore` and `package.json` to include new dependencies.

### Detailed summary
- Updated `import` statements to use `@sinclair/typebox` for type definitions.
- Made `inputs` and `outputs` in `abiSchema` optional.
- Added a `biome.json` configuration file for project settings.
- Enhanced response schemas in `readContract` and `writeToContract` functions.
- Updated `.gitignore` to include `.aider*` files.
- Added `@biomejs/biome` as a dev dependency in `package.json`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/server/schemas/contract/index.ts [hunk 1]
```diff
@@ -17,6 +17,9 @@ export const readRequestQuerySchema = Type.Object({
   ),
 });
 
+//make the type interface support a generic dictionary
+export interface readResponseSchema {}
```
Comment (arcoraven) [inline src/server/schemas/contract/index.ts:21]: This looks unused.

File: src/server/routes/contract/write/write.ts [hunk 2]
```diff
@@ -26,19 +26,35 @@ const writeRequestBodySchema = Type.Object({
...
-      Type.Array(Type.Any()),
-      Type.Any(),
-    ]),
-  ),
+  args: Type.Array(Type.Any(), {
+    description: "The arguments to call on the function",
+  }),
   ...txOverridesWithValueSchema.properties,
-  abi: Type.Optional(Type.Array(abiSchema)),
+  abi: Type.Optional(Type.Array(Type.Object({
```
Comment (arcoraven) [inline src/server/routes/contract/write/write.ts:33]: Is this different than abiSchema below (in `src/server/schemas/contract/index.ts`)? Hard to tell but it looks the same at a glance.

Comment (arcoraven) [reply src/server/routes/contract/write/write.ts:33]: Looks the same as I walk through it.

**Comment by arcoraven** [comment]
Tested locally:
- contract read
- contract write w/ autodetecting ABI
- contract write w/ provided ABI

My main concern is the build is erroring on that `server.register()` line because it doesn't recognize `refResolver`?

**Review by arcoraven — APPROVED** [review]

**Comment by arcoraven** [comment]
But the explorer page and the `/json` endpoint clearly work so...the typescript error just seems wrong? Going to merge and confirm if the Docker image builds properly.

**Comment by furqanrydhan** [comment]
Went through the spec now, still passes on everything. If we can run tests + use it I can use this now for my next project! The openapi spec was a dependency

**Pull request merged**